clusterloader2/docs/test_suite_design.md

```diff
@@ -0,0 +1,40 @@
+# Cluster loader - TestSuite
+
+## Background
+
+Since 2019 cluster loader become main tool for kubernetes e2e performance tests.
+It's very convenient for running single test case, but currently it's impossible
+to run the same test with two different sets of parameters in a single job run.
+This forces developers to create separate job configs for almost each test.
+
+## TestSuite
+
+
+```
+type TestSuite []TestScenario
+
+type TestScenario struct {
+	// Identifier is a unique test scenario name across test suite.
+	Identifier string
+	// ConfigPath defines path to the config file that satisfy already
+        // existing cluster loader API.
+	ConfigPath string
+	// OverridePaths defines what override files should be applied
+        // to the config specified by the ConfigPath.
+	OverridePaths []string
+}
+```
+
+Test scenario level overrides benefits:
+- Ability to use the same template with multiple overrides in a single job.
+- Less error prone - variables with same names across different configs
+could have been accidentally overwritten before.
+
+To make using same test configs in one test suite more user-friendly,
+the Identifier field was introduced. This identifier will be used to match
+run results to the corresponding test.
+
+## Other
+
+After TestSuite is support by cluster loader, passing config paths and overrides
+using flags will be disabled.
```

Comment:
NIT: After TestSuite is **supported**

Also, what does it mean in terms of passing testsuite to the clusterloader?  Where the discussion landed? Will there be a new flag for testsuite?
Could you clarify it here so it's clear what we should expect?

Reply:
My personal preference would be that:
- we define yamls in test-infra (because they basically describe job config)
- we pass path to yaml to job config

Reply:
> we pass path to yaml to job config

+1

> we define yamls in test-infra (because they basically describe job config)

My assumption is that if test config changes, sometimes you want to update test suite in the same PR. IMHO configs should be in perf-tests.

Reply:
As discussed offline - this should live in test-infra, because that doesn't define test - it's only defining job config.

Reply:
Please mention that the supported way will be passing (via flag) a path to config file with TestSuite.

Reply:
Done.